### PR TITLE
Discord OSINT tool added

### PR DESCRIPTION
**Pull request by rachejazz**
```
┬─[divya at racharch in ~/a/g/OWASP-Maryam]─[G:discord]─[B:28%, 00:25:25 remaining]
╰──> λ ./maryam -e discord_search -s OSINT
[*] [DISCORD] Searching in 0 page...
[*] [DISCORD] Searching in 1 page...
[*] [DISCORD] Searching in 2 page...
[*] SERVERS
[*] 	Frenchhub
 Join: 	https://disboard.org/server/join/812333805926481980
[*] 	Global issues
 Join: 	https://disboard.org/server/join/655512137200041984
[*] 	[Little] Think Tank
 Join: 	https://disboard.org/server/join/624955124824932372
[*] 	global issues
 Join: 	https://disboard.org/server/join/655511421546921994
[*] 	Mirage | Kaneki
 Join: 	https://disboard.org/server/join/808415611830140970
[*] 	Hack Talk
 Join: 	https://disboard.org/server/join/743831555767795812
[*] 	Team Omega Cybersecurity and Analysis
 Join: 	https://disboard.org/server/join/605819996546924544
[*] 	EmailTools
 Join: 	https://disboard.org/server/join/797162096021602305
[*] 	Projet FOX
 Join: 	https://disboard.org/server/join/775480337727225879
```
```
┬─[divya at racharch in ~/a/g/OWASP-Maryam]─[G:discord]─[B:28%, 00:25:07 remaining]
╰──> λ ./maryam -e discord_search -u 34
[*] [DISCORD] Searching in 0 page...
[*] [DISCORD] Searching in 1 page...
[*] [DISCORD] Searching in 2 page...
[*] SERVERS
[*] Without result
[*] USERS
[*] 	Oura#3403
 Check Full Profile: 	https://discord.id/?prefill=69938124834213888
[*] 	JesusPT#3410
 Check Full Profile: 	https://discord.id/?prefill=74256780116373504
[*] 	Apollo#3475
 Check Full Profile: 	https://discord.id/?prefill=72148988215697408
[*] 	Asutorea#3428
 Check Full Profile: 	https://discord.id/?prefill=66544817978486784
[*] 	開箱窮三代#3465
 Check Full Profile: 	https://discord.id/?prefill=67404795220078592
[*] 	Entrench#3421
 Check Full Profile: 	https://discord.id/?prefill=70213673167093760
[*] 	Budi#3426
 Check Full Profile: 	https://discord.id/?prefill=74821294465486848
[*] 	Eclipsez#3481
 Check Full Profile: 	https://discord.id/?prefill=76572806023749632
[*] 	kokalo#3456
 Check Full Profile: 	https://discord.id/?prefill=73787567899680768
[*] 	BeautifulCleanCoal#3467
 Check Full Profile: 	https://discord.id/?prefill=67455031992918016
[*] 	Tyrsfall#3412
 Check Full Profile: 	https://discord.id/?prefill=66903565599244288
[*] 	degede#3457
```

### CODE REVIEW

**Comment by rachejazz** [comment]
#135

File: core/util/discord.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+			urls = url_s
+		if self.q.split('_')[1] == 'u':
+			max_attempt = len(url_u)
+			urls = url_u
+		for url in range(max_attempt):
+			self.framework.verbose(f"[DISCORD] Searching in {url} page...")
+			try:
+				req = self.framework.request(url=urls[url], allow_redirects=True)
+			except Exception as e:
+				print(req.text)
```
Comment (saeeddhqan) [inline core/util/discord.py]: ?

File: modules/osint/discord_search.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	if 'limit' in varnames:
+		attr = engine(q, limit)
+	attr.run_crawl()
+	PAGES += attr.pages
+
+def module_api(self):
+	user = self.options['user']
+	server = self.options['server']
+	if user and server:
+		self.error("Choose to search only one among user and server.")
```
Comment (saeeddhqan) [inline modules/osint/discord_search.py:52]: It won't show! because you are in module_api

File: modules/osint/discord_search.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+		return
+	elif user:
+		if user.isdigit():
+			query = f'{user}_u'
+		else:
+			self.error("Enter user's 4 digit tag only [ #<Number> ]")
```
Comment (saeeddhqan) [inline modules/osint/discord_search.py:58]: as well. You can only show the errors in module_run. Or returning something like this:
{'error': 'msg'}

File: modules/osint/discord_search.py (hunk 2 above)
Comment (rachejazz) [reply modules/osint/discord_search.py:52]:
```
┬─[divya at racharch in ~/a/g/OWASP-Maryam]─[G:discord]
╰──> λ ./maryam -e discord_search -u 34 -s Wolf
[!] Choose to search only one among user and server.
[*] None
```

File: modules/osint/discord_search.py (hunk 3 above)
Comment (rachejazz) [reply modules/osint/discord_search.py:58]:
```
┬─[divya at racharch in ~/a/g/OWASP-Maryam]─[G:discord]
╰──> λ ./maryam -e discord_search -u wolf
[!] Enter user's 4 digit tag only [ #<Number> ].
[*] None
```

File: core/util/discord.py (hunk 1 above)
Comment (rachejazz) [reply core/util/discord.py]: Removed

**Comment by rachejazz** [comment]
@saeeddhqan do let me know if there's any further problems. Everything works fine for me

**Comment by saeeddhqan** [comment]
I'll check it out.
Thank you!

**Pull request merged**